### PR TITLE
fixes #1049: path segment search with in-memory filter fixed

### LINKED ISSUE #1049: Document search with path segment and $in returns wrong results
Document search with `where={"n,m.value": {"$in": [5]}}` will match both `{"n": { "value": 5}}` and `{"m": { "value": 8}}`. Path segment is not considered when doing the row matches filter path check.

### PR DESCRIPTION
**Pull request by ivansenic**
See #1049 for details. Two fixes:

- `convertArrayPath` to transform also the path segments and convert directly to expected path
   - this enables complete removal of calls to the `convertArrayPath` outside the expression parsing
- filter expression to consider path segments when evaluating

Not sure if this was working before, there were no tests.

### CODE REVIEW

File: restapi/src/test/java/io/stargate/web/docsapi/service/query/FilterExpressionTest.java [hunk 1]
```diff
@@ -342,6 +342,32 @@ public void conditionFalse() {
       verifyNoMoreInteractions(condition);
     }
 
+    @Test
+    public void conditionFalsePathSegment() {
```
Comment (dimas-b) [inline restapi/src/test/java/io/stargate/web/docsapi/service/query/FilterExpressionTest.java:346]: Do we have a unit test for a positive segment match? Sorry, I could not find it.

File: restapi/src/main/java/io/stargate/web/docsapi/service/query/DocumentServiceUtils.java [hunk 2]
```diff
@@ -32,7 +34,30 @@
...
+   * Converts given path to an array one if needed:
+   *
+   * <ol>
+   *   <li>[1] - converted to [000001]
+   *   <li>[0],[1] - converted to [000000],[000001]
+   *   <li>[*] - not converted
+   *   <li>anyPath - not converted
+   * </ol>
+   *
+   * @param path filter or filed path
```
Comment (EricBorczuk) [inline restapi/src/main/java/io/stargate/web/docsapi/service/query/DocumentServiceUtils.java:47]: nit: `filed path` -> `field path`

File: restapi/src/main/java/io/stargate/web/docsapi/service/query/FilterExpression.java [hunk 3]
```diff
@@ -161,8 +162,16 @@ private boolean matchesFilterPath(Row row) {
         return false;
       }
 
-      // if not equal, fail
-      if (!Objects.equals(path, target)) {
+      boolean pathSegment = target.contains(",");
+      // if we have the path segment, we need to check if any matches
+      if (pathSegment) {
```
Comment (EricBorczuk) [inline restapi/src/main/java/io/stargate/web/docsapi/service/query/FilterExpression.java:167]: :+1:

**Review by dimas-b — APPROVED** [review]

File: testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java [hunk 4]
```diff
@@ -1778,6 +1778,30 @@ public void searchPathSegmentMatching() throws Exception {
...
+    String r =
+        RestUtils.get(
+            authToken,
+            hostWithPort
+                + "/v2/namespaces/"
+                + keyspace
+                + "/collections/collection?page-size=3&where={\"n,m.value\": {\"$in\": [5]}}&raw=true",
+            200);
+
+    String expected = "{\"matching1\":{\"n\":{\"value\":5}}}";
```
Comment (EricBorczuk) [inline testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java:1801]: :+1: thanks for adding this case!

**Review by EricBorczuk — APPROVED** [review]
This looks good to me! Thanks for fixing and adding test cases

File: restapi/src/test/java/io/stargate/web/docsapi/service/query/FilterExpressionTest.java (hunk 1 above)
Comment (ivansenic) [reply restapi/src/test/java/io/stargate/web/docsapi/service/query/FilterExpressionTest.java:346]: Yea there is no need for this, we wanted to makes sure here that the path match, and I can not do that positive condition.

Comment (dimas-b) [reply restapi/src/test/java/io/stargate/web/docsapi/service/query/FilterExpressionTest.java:346]: ok :+1:

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**